packages/tldraw/src/lib/ui/context/actions.tsx

```diff
@@ -65,6 +67,17 @@ export interface ActionsProviderProps {
 	children: React.ReactNode
 }
 
+/** @public */
+export function supportsDownloadingOriginal(
+	shape: TLShape,
+	editor: Editor
+): shape is TLImageShape | TLVideoShape {
+	return (
+		(editor.isShapeOfType(shape, 'image') || editor.isShapeOfType(shape, 'video')) &&
+		!!(shape as any).props.assetId
```

Comment:
my API head goes to `ShapeUtil.getDownload` or something, where the downloadable asset is generated. This would be really useful for example in tldraw computer for images / audio / videos, which we've had to implement separately.

Reply:
I wouldn't rely too much on the assets, that seems like an implementation detail. Our asset system is really only needed if you're trying to minimize duplication (such as with base64'd image sources) or if you expect the asset's source to change and need to react to that change. For example in computer the shapes just have the url of the source image / video. There could be lots of third party shapes where the asset isn't stored in our asset system.

Reply:
I don't think this is a blocker, I'd like to have this action in there (I separately implemented it in my branch of mime's image toolbar PR